### DISCUSSION TITLE
Polymer

### DISCUSSION THREAD

**Discussion by fredericksilva** [frontendbr/forum#588] [Perguntas e Respostas]
Galera estou montando uma grade de estudo sobre polymer, quais assuntos vocês acham que não deve faltar nessa grade? 

Obrigado a todos!

**Comment by felipefialho** [comment]
@afonsopacifer Você que manja disso?

**Comment by bernardodiasc** [comment]
Eu diria Web Components, mas não somente a parte técnica do HTML5, diria o conceito como um todo. A maioria dos frameworks atuais trabalham com esses conceitos de componente web, que seria um conjunto de código HTML, CSS e JS (ou outras linguagens que convertam pra essas), usando as regras do framework então resultando alguma parte visual, ou mesmo comportamental que atua na página. Os sonhos molhados de todo desenvolvedor web é ter componentes que se encaixam lindamente resultando num site bacanudo, acho que a maioria dos frameworks de frontend atuais visam isso e prometem "ease of use" de diferentes formas. Confesso que só li a respeito de Polymer, não usei de fato, então minhas sugestões param por aqui :)

**Comment by afonsopacifer** [comment]
Bem, os principais links com tudo o que você precisar saber e algumas infos importantes estão ai:

> ps: Polymer está mudando para a versão 2, mas não esquenta que ele tem um modo híbrido para compatibilidade com os componentes criados com a versão 1.

Sobre o que está acontecendo com a nova versão, e o porque ela é revolucionaria eu escrevi no artigo [Polymer 2 e o futuro onde os Web Components dominam a terra.](https://medium.com/@afonsopacifer/polymer-2-e-o-futuro-onde-os-web-components-dominam-a-terra-308a42858470)

Para aprender sobre, tem o screencast do Google sobre o assunto:
https://www.youtube.com/watch?v=y2FxAT1u8pU&list=PLNYkxOF6rcIDdS7HWIC_BYRunV6MHs5xo

Também tem videos de talks do time do Polymer em edições do Google I/O 2016:
https://www.youtube.com/watch?v=fFF2Yup2dMM&list=PLNYkxOF6rcIDnSm7bZRJC36Ca1DYXSQ70

e Google I/O 2017: (não achei a Playlist, mas com esse link vc acha tudo)
https://www.youtube.com/watch?v=assSM3rlvZ8&t

O Polymer tem seu próprio evento tbm, chamado Polymer Summit!

Summit 2015:
https://www.youtube.com/watch?v=PbYvv2VUp90&list=PLNYkxOF6rcICdISJclfQhj2S8QZGjXV8J

Summit 2016:
https://www.youtube.com/watch?v=0iM0DZjYGqg&list=PLNYkxOF6rcICc687SxHQRuo9TVNOJelSZ

Para conteúdo em português, tem minha série de videos:
https://www.youtube.com/playlist?list=PLgcWRW66amTkWM7E2vdw3lMoDSAJG7vdl

Também tem um Hangout que fiz para o GDG-BH sobre o assunto:
https://www.youtube.com/watch?v=AfOis1W-pkU&t

Tem outro Hangout do GDG Maceió com o Felipe Sousa:
https://www.youtube.com/watch?v=uxZElHIPcek

Além disso, tem a melhor opção que é a documentação no site oficial:
https://www.polymer-project.org/

Por fim, fiz uma checklist para criar componentes chamada "Good Practices for build Web Components", com ela vc consegue pegar os itens que precisa estudar ❤️ 
https://github.com/hippie-elements/web-components-good-practices

É isso, rsrsr, se tiver alguma dúvida sobre o que estiver estudando, só mandar  :)

**Comment by grassiricardo** [comment]
boa @afonsopacifer, estou seguindo e aplicando o conteúdo que você disponibiliza em seu canal, o assunto sobre polymer está ótimo!

**Comment by yanmagale** [comment]
@fredericksilva nessa super lista que o @afonsopacifer mandou, eu so incluiria a serie sobre Polymer feita pelo Rob Dodson - [Polycasts](https://www.youtube.com/playlist?list=PLOU2XLYxmsII5c3Mgw6fNYCzaWrsM3sMN)

Sobre um "guia de estudos", eu incluiria:
- Javascript/ES6
- Web Components
- Ciclo de Desenvolvimento de um App Polymer
- PRPL

**Comment by yanmagale** [comment]
Recentemente eu participei no evento GDG Tech Tour em Caruaru, falando sobre o Polymer 2:
- Link dos slides -  https://t.co/g0I2Oj9k3F
- Link dos Exemplos mostrados - https://github.com/yanmagale/polymer-talk-tech-tour-caruaru
